vault/resource_aws_secret_backend.go

```diff
@@ -143,12 +150,14 @@ func awsSecretBackendCreate(d *schema.ResourceData, meta interface{}) error {
 	iamEndpoint := d.Get("iam_endpoint").(string)
 	stsEndpoint := d.Get("sts_endpoint").(string)
 	usernameTemplate := d.Get("username_template").(string)
+	local := d.Get("local").(bool)
 
 	d.Partial(true)
 	log.Printf("[DEBUG] Mounting AWS backend at %q", path)
 	err := client.Sys().Mount(path, &api.MountInput{
 		Type:        consts.MountTypeAWS,
 		Description: description,
+		Local:       local,
```

Comment:
Looking great! We should also set this value to state if we read from Vault. We can add the following in the read method as done on [these lines here](https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_aws_secret_backend.go#L258-L261)

```
d.Set("local", mount.Local)
```